compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -3507,7 +3507,8 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
             }
             ObligationCauseCode::OpaqueReturnType(expr_info) => {
                 if let Some((expr_ty, expr_span)) = expr_info {
-                    let expr_ty = with_forced_trimmed_paths!(self.ty_to_string(expr_ty));
+                    let expr_ty =
+                        with_forced_trimmed_paths!(self.tcx.short_ty_string(expr_ty, &mut None));
```

Comment:
I think we can remove the macro annotation if we're using `short_ty_string` already:

```suggestion
                    let expr_ty = self.tcx.short_ty_string(expr_ty, &mut None);
```

Reply:
(The other comment about `&mut None` applies here too.)